.github/workflows/opex_api_session_manager.yaml

```diff
@@ -0,0 +1,105 @@
+name: Opex Dashboard session-manager
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "apps/session-manager/api/*.yaml"
+      - ".github/workflows/opex_api_session_manager.yaml"
+      - "apps/session-manager/.opex/**"
+
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}
+
+jobs:
+  filter:
+    runs-on: ubuntu-22.04
+    permissions:
+      pull-requests: read
+    outputs:
+      bpd: ${{ steps.filter.outputs.bpd }}
+      fast-login: ${{ steps.filter.outputs.fast-login }}
+      fims: ${{ steps.filter.outputs.fims }}
+      internal: ${{ steps.filter.outputs.internal }}
+      pagopa: ${{ steps.filter.outputs.pagopa }}
+      public: ${{ steps.filter.outputs.public }}
+      zendesk: ${{ steps.filter.outputs.zendesk }}
+    # Checks to see if openapi specs in the PR match one of the ones listed.
+    # We can use the result to decide which job to run
+    # With this type of filter we can avoid creating a lot of workflow files
+    steps:
+      # path-filters action requires checkout with push event
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 20
```

Comment:
Why is it different from the `pr_opex_api_session_manager.yaml` one?

Reply:
based on the [official github action example for the feature branch](), the push on branch filter needs this to prevent a bug:

```yaml
steps:
    - uses: actions/checkout@v4
      with:
        # This may save additional git fetch roundtrip if
        # merge-base is found within latest 20 commits
        fetch-depth: 20
    - uses: dorny/paths-filter@v3
```

instead, the pr action doesn't need it